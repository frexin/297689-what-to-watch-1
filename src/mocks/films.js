const movies = [
  {
    id: 1, name: `Bohemian Rhapsody`, picture: `bohemian-rhapsody.jpg`, genre: `Crime`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 2, name: `Aviator`, picture: `aviator.jpg`, genre: `Dramas`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`

  },
  {
    id: 3, name: `What We Do in the Shadows`, picture: `what-we-do-in-the-shadows.jpg`, genre: `Crime`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 4, name: `Snatch`, picture: `snatch.jpg`, genre: `Horror`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    id: 5, name: `Revenant`, picture: `revenant.jpg`, genre: `Romance`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

export {movies};
