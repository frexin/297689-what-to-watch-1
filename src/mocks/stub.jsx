import React from "react";

const StubComponent = () => {
  return (<div/>);
};

export default StubComponent;
