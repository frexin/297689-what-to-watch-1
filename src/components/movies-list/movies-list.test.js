import React from 'react';
import renderer from 'react-test-renderer';

import {movies} from "../../mocks/films";
import MoviesList from "./movies-list";
import {BrowserRouter} from "react-router-dom";

it(`MoviesList component loads correctly`, () => {
  const tree = renderer.create(<BrowserRouter><MoviesList hasMoreMovies={false} onLoadMore={jest.fn()} movies={movies} /></BrowserRouter>).toJSON();

  expect(tree).toMatchSnapshot();
});
