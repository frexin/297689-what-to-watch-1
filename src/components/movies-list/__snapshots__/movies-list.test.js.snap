// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList component loads correctly 1`] = `
<div
  className="page-content catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Bohemian Rhapsody"
        height={175}
        src="bohemian-rhapsody.jpg"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Aviator"
        height={175}
        src="aviator.jpg"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
      >
        Aviator
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="What We Do in the Shadows"
        height={175}
        src="what-we-do-in-the-shadows.jpg"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
      >
        What We Do in the Shadows
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Snatch"
        height={175}
        src="snatch.jpg"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
      >
        Snatch
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Revenant"
        height={175}
        src="revenant.jpg"
        width={280}
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
      >
        Revenant
      </a>
    </h3>
  </article>
</div>
`;
